templates/zerver/api/roles-and-permissions.md

```diff
@@ -0,0 +1,85 @@
+that return user account data include a `role` property, which
+contains the user's organization-level role. These roles are encoded as:
+
+* Organization owner: 100
+
+* Organization administrator: 200
+
+* Organization moderator: 300
+
+* Member: 400
+
+* Guest: 600
+
+User account data also include these relevant boolean properties:
+
+* `is_admin` specifying whether the user is an organization administrator.
+
+* `is_guest` specifying whether the user is a guest user.
+
+* `is_owner` specifying whether the user is an organization owner, and
+  present as of [Zulip 3.0](/api/changelog#changes-in-zulip-30) (feature
+  levels 8 and 11). If true, `is_admin` will also be true.
+
+* `is_billing_admin` specifying whether the user is a billing administrator,
+  and present as of [Zulip 5.0](/api/changelog#changes-in-zulip-50)
+  (feature level 73).
+
+Additionally, as of [Zulip 4.0](/api/changelog#changes-in-zulip-40)
+(feature level 60), when `realm_user` is present in `fetch_event_types`,
+the [`/register`endpoint](/api/register-queue) will return an
+`is_moderator` boolean value specifying whether the user is an organization
+moderator.
+
+Note that the [`GET users/me`](/api/get-own-user) endpoint did not
+return a `is_guest` value until
+[Zulip 3.0](/api/changelog#changes-in-zulip-30) (feature level 10).
+
+## Permissions levels
+
+Many areas of Zulip are customizable by the organization-level roles
+above, such as (but not limited to) [configuring message editing and
+deletion](/help/configure-message-editing-and-deletion) and
+[streams permissions](/help/stream-permissions).
+
+These permission levels and policies in the API are designed to be
+cutoffs in that users with the specified role and above have the
+specified ability or access.
+
+The most trusted, and therefore most restrictive, permission level or policy
+would be organization owner only, or nobody. And the least trusted, and
+therefore least restrictive, permission level would be members only,
+or everyone.
+
+## Determining if a user is a full member
+
+When a Zulip organization has set up a [waiting period before new members
+turn into full members](/help/restrict-permissions-of-new-members),
+clients will need to determine if a user's account has aged past the
+organization's waiting period threshold.
+
+The `realm_waiting_period_threshold`, which is the number of days until
+a user's account is treated as a full member, is returned by the
+[`/register` endpoint](/api/register-queue) when `realm` is present in
+`fetch_event_types`.
+
+Clients can compare the `realm_waiting_period_threshold` to a user
+accounts's `date_joined` property, which is the time the user account
+was created, to determine if a user has the permissions of a full member
+or a new member.
+
+Note that the [`GET users/me`](/api/get-own-user) endpoint did not
+return a `date_joined` value until
+[Zulip 3.0](/api/changelog#changes-in-zulip-30) (feature level 10).
```

Comment:
I think this can also be left unmentioned here, and something to find on the endpoint doc when it's relevant.

(For a full-blown client, it won't be relevant because they're getting the user data from the event system with /api/register-queue, not from separate fetches like /api/get-own-user. For a simple bot, they're unlikely to worry about supporting ancient versions.)

Reply:
Yeah, I agree with this.